### PR TITLE
Add property for easy entity orientation.

### PR DESCRIPTION
**Pull request by mramato**
1. Add `OrientDirectionOfTravelProperty`, which takes a position property and generates a `Quaternion` along the direction of travel. Thanks to @bagnell for providing me the code to actually computes the orientation.
2. Added a new Sandcastle example, which is a copy of the tie-figter demo that now uses the Cesium aircraft and the new orientation property. 
3. Added ability to pass an initial array of intervals to TimeIntervalCollection, which makes constructing entities with availability easier.

I'm up for better naming suggestions if anyone has one, but I don't want to start arguing over it, let me know if you really hate the current name and have an objectively better one.

Also, while test coverage for the new class is 100%, I have no idea how to test the actual results because I don't know of any situations that would provide "nice" numbers.  I admit for the current numbers part I just copied values from the results. Does anyone have any ideas here?

### CODE REVIEW

**Comment by speigg** [comment]
How about `VelocityDirectedOrientation` ?

**Comment by mramato** [comment]
I don't know why that didn't occur to me. I think I'm going to go with the slightly shorter `VelocityOrientationProperty`  Thanks for the suggestion.

**Comment by mramato** [comment]
Okay, rename is done and this is ready to be reviewed.

File: Source/DataSources/VelocityOrientationProperty.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        if (!defined(position2)) {
+            position2 = position1;
+            position1 = property.getValue(JulianDate.addSeconds(time, -step, timeScratch), position2Scratch);
+
+            if (!defined(position1)) {
+                return undefined;
+            }
+        }
+
+        var normal = defaultValue(this._ellipsoid, Ellipsoid.WGS84).geodeticSurfaceNormal(position1, normalScratch);
```
Comment (pjcozzi) [inline Source/DataSources/VelocityOrientationProperty.js]: Can we move this code to a generic function (could even be `@private` for now) that given the ellipsoid, position, and vector, creates the 3x3 matrix?  Perhaps it belongs in [Transforms](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Transforms.js).  This is a farily common operation.

Also, are we sure this robustly handles all possible velocities, e.g., when the object is traveling along the surface normal?  I used the same algorithm for Santa one year and needed a special case for `Cartesian3.cross(direction, normal, rightScratch)`.

**Comment by pjcozzi** [comment]
Can we scale up the model in the Sandcastle example?  At first, I didn't even see it.

File: Source/DataSources/VelocityOrientationProperty.js (hunk 1 above)
Comment (mramato) [inline Source/DataSources/VelocityOrientationProperty.js]: Okay, I refactored out the code into `Transforms.rotationMatrixFromPositionVelocity`.  I see what you mean about needing the special case (though I didn't make any code changes to handle it myself).  @bagnell can you take a quick look?

**Comment by mramato** [comment]
> Can we scale up the model in the Sandcastle example? At first, I didn't even see it.

I bumped up the min pixel size to 64.

**Comment by pjcozzi** [comment]
@bagnell can you look at this?

> Also, are we sure this robustly handles all possible velocities, e.g., when the object is traveling along the surface normal? I used the same algorithm for Santa one year and needed a special case for Cartesian3.cross(direction, normal, rightScratch).

It would be nice for this to make 1.8.

**Comment by bagnell** [comment]
@pjcozzi I handled the case where the cross product could be zero. The rest looks good to me. You can merge when you're happy.

**Comment by pjcozzi** [comment]
Looks good, thanks.

**Pull request merged**